### PR TITLE
Re-map light brightness range

### PR DESCRIPTION
**Pull request by postlund**
Home Assistant uses brightness range 0-255 whilst Tuya somewhere around
29-1092. This commit maps brightness between these two ranges.

### CODE REVIEW

**Comment by postlund** [comment]
@rospogrigio I think this is ready, but I want SmartM-ui (forums) to try it out with #62 before merging. But feel free to review.

File: custom_components/localtuya/light.py
Comment (rospogrigio) [inline custom_components/localtuya/light.py:38]: Why not `round` instead of `int`?

**Review by rospogrigio — CHANGES_REQUESTED** [review]
And you too should update the YAML samples in `__init__.py`  😄

**Comment by postlund** [comment]
> And you too should update the YAML samples in `__init__.py` 😄

Lol, yeah 😉

File: custom_components/localtuya/light.py
Comment (postlund) [reply custom_components/localtuya/light.py]: Only n00bz use `round`. So yeah, let's use `round`!

**Comment by postlund** [comment]
Should be fixed now!

**Comment by SmartM-ui** [comment]
Here I am!
Nice to read you, I almost worry when you don't answer right away :-)

I installed version # 80, restarted Home Assistant, put this code in my yaml file and restarted HA

localtuya:
   - host: 192.168.1.131 #lampadario studio
     device_id: 6xxxxxxd
     local_key: bxxxxxxxx9
     friendly_name: Lampadario Studio LocalTuya
     protocol_version: "3.3"
     entities:
       - platform: light
         friendly_name: Lampadario Studio LocalTuya
         id: 20
         brightness: 22
         color_temp: 23
         brightness_lower: 10
         brightness_upper: 1000

Unfortunately there is still the problem that it fails to turn on the light bulb and after 2 seconds the status returns to off and in any case it cannot manage even the brightness.

Log:
2020-10-15 20:22:14 DEBUG (MainThread) [custom_components.localtuya.pytuya] [6xxxxxxxxxxxxxd] Sending command heartbeat (device type: type_0a)
2020-10-15 20:22:14 DEBUG (MainThread) [custom_components.localtuya.pytuya] [6xxxxxxxxxxxxxd] Paylod: b'{}'
2020-10-15 20:22:14 DEBUG (MainThread) [custom_components.localtuya.pytuya] [6xxxxxxxxxxxxxd] Waiting for sequence number -100
2020-10-15 20:22:14 DEBUG (MainThread) [custom_components.localtuya.pytuya] [6xxxxxxxxxxxxxd] Got heartbeat response
2020-10-15 20:22:14 DEBUG (MainThread) [custom_components.localtuya.pytuya] [6xxxxxxxxxxxxxd] Decode payload: b''
2020-10-15 20:22:14 DEBUG (MainThread) [custom_components.localtuya.pytuya] [6xxxxxxxxxxxxxd] Decrypted payload: {}

**Comment by postlund** [comment]
Oh, I screwed up... Gimmie a moment to fix that.

**Comment by postlund** [comment]
I have updated the PR now, try updating and run again. Should be smooth sailing now ⛵️

**Comment by SmartM-ui** [comment]
> I have updated the PR now, try updating and run again. Should be smooth sailing now ⛵️

I'm very happy.
You did great and you outdid yourself
Now the brightness is managed correctly and even the on off status has no more problems.
Great news in this version that surprised me ...
The LocalTuya app can coexist with the SmartLife app and work in real time! WOW
Pierre I Love You !!!! :-)

**Comment by postlund** [comment]
That's great @SmartM-ui, thanks for verifying 😀 I guess this PR is ready for review @rospogrigio.

File: custom_components/localtuya/light.py
Comment (rospogrigio) [inline custom_components/localtuya/light.py:68]: What happens here if CONF_COLOR_TEMP is not configured? Is there a default value to return?

**Review by rospogrigio — CHANGES_REQUESTED** [review]

File: custom_components/localtuya/light.py
Comment (postlund) [reply custom_components/localtuya/light.py:68]: The default value is to return `None`, but I should probably add an explicit return for it.

**Review by rospogrigio — APPROVED** [review]

**Comment by rospogrigio** [comment]
OK, then. 
Feel free to merge if you think it's ready (don't know if you want to wait for feedback from accelle17 for Nebula lights first).

**Comment by postlund** [comment]
It has been confirmed to work before, so we can take it as a bug if it doesn't work.

**Comment by SmartM-ui** [comment]
Hi @postlund @rospogrigio , 
I found a different value between SmartLife app and LocalTuya regarding the color temp.
Would it be possible to choose the right range for your bulb so that you can have the color temperature gradations?

Here are some examples:

Center of the slider on LocalTuya (color temp 258) corresponds to 13% on Smartlife app (instead it should be 50%)

other test:
SmartLife color temp: 0%
LocalTuya color_temp: 370

SmartLife color temp: 10%
LocalTuya color_temp: 284

SmartLife color temp: 50%
LocalTuya color_temp: -57

SmartLife color temp: 100%
LocalTuya color temp: -470

Hi, I found a different value between SmartLife app and LocalTuya regarding the color temperatyra.
Would it be possible to choose the right range for your bulb so that you can have the color temperature gradations?

Here are some examples:

Center of the slider on LocalTuya (color temp 258) corresponds to 13% on Smartlife app instead it should be 50%

How can I do to open a new request for ColorTemp?

Thanks

**Comment by postlund** [comment]
Yeah, this should be fixed as well. We have some hardcoded color temperatures in our implementation. Ideally lower and upper color temperature should be specified in Kelvin. Should be a simple task if someone has some time (I'm trying to figure out RGB at the moment).

**Comment by SmartM-ui** [comment]
Well, focus on RGB, much more useful!
This feature would make LocalTuya the ultimate Tuya device management app.
Hope you can find a way to process RGB colors.
If I can help with the tests, I'm available

Have a nice day

You will insert the news regarding the RGB work progress in the topic: https://github.com/rospogrigio/localtuya-homeassistant/issues/33 ?

**Pull request merged**
